### PR TITLE
Add new fluent functions for words that change based on gender

### PR DESCRIPTION
**Pull request by mirrorcult**
Needed for and requires space-wizards/space-station-14#4071

### CODE REVIEW

**Comment by RemieRichards** [comment]
A lot of these smell remarkably... "englishy"

**Comment by mirrorcult** [comment]
what exactly is the alternative? other localizations will use different .ftl files and so won't need these functions and will write their own for the language they're localizing for

**Comment by RemieRichards** [comment]
Exactly, they'll use their own ftl files, which ideally wouldn't have access to such locale-specific functions.

I wonder if when registering functions we can annotate what locales its intended for, and have it simply not work/return a warning when used in "invalid" locales. (For example, not all languages will need or want to implement a `PROPER()`)

It would've been nice to implement `THE()` and `PROPER()` and so on in ftl directly, as terms, but sadly you can't pass entities around as arguments inside of fluent itself (only from C#->ftl)

**Comment by mirrorcult** [comment]
locale-specific functions would work for this, yeah, but its out of scope as far as im concerned (unless its very easy in which case i will do it)

**Comment by RemieRichards** [comment]
I think it's both. (out of scope of this PR, and easy), so up to you.

It'd just be nice to curtail it before we have like 50 bajillion functions (this pr *does* increase the number of functions by a lot, which is what got me thinking, is all)

File: Robust.Shared/Localization/LocalizationManager.Functions.cs [hunk 1]
```diff
@@ -1,33 +1,63 @@
...
             return new LocValueString(GetString("zzzz-the", ("ent", args.Args[0])));
         }
 
+        /// <summary>
+        /// Returns the string passed in, with the first letter capitalized.
+        /// </summary>
+        private ILocValue FuncCapitalize(LocArgs args)
+        {
+            var input = args.Args[0].Format(new LocContext());
+            System.Console.WriteLine(input);
```
Comment (PaulRitter) [inline Robust.Shared/Localization/LocalizationManager.Functions.cs]: grafik

**Review by PaulRitter — DISMISSED** [review]

File: Robust.Shared/Localization/LocalizationManager.Functions.cs (hunk 1 above)
Comment (mirrorcult) [reply Robust.Shared/Localization/LocalizationManager.Functions.cs]: oops

**Review by ShadowCommander — APPROVED** [review]

**Pull request merged**